src/python/grpcio/grpc/beta/implementations.py

```diff
@@ -105,10 +106,37 @@ def metadata_call_credentials(metadata_plugin, name=None):
     A CallCredentials object for use in a GRPCCallOptions object.
   """
   if name is None:
-    name = metadata_plugin.__name__
```

Comment:
It seems to me that an object's name, if present, is more likely to be helpful than its class name. Should this code look for the object name and only use the class name if it is not found? I believe @soltanmm originally authored this and may have an opinion.